src/main/scala/net/snowflake/spark/snowflake/SnowflakeRelation.scala

```diff
@@ -201,6 +204,112 @@ private[snowflake] case class SnowflakeRelation(
+        totalCompressedSize += firstChunkSize
+        totalUnCompressedSize += firstChunkSize
+        if (firstRowCount > 0) {
+          log.info(
+            s"""printStatForSnowflakeResultSetRDD First result chunk:
+               | rowCount=$firstRowCount chunkSize=$firstChunkSize
+               |""".stripMargin.filter(_ >= ' '))
+        }
+
+        // Get row count for all pre-signed URLs from metadata
+        var index = 0
+        val fileChunks = resultSet.asInstanceOf[SnowflakeResultSetSerializableV1]
+          .getChunkFileMetadatas
+        while (index < fileChunks.size) {
+          totalRowCount += fileChunks.get(index).getRowCount
+          totalCompressedSize += fileChunks.get(index).getCompressedByteSize
+          totalUnCompressedSize += fileChunks.get(index).getUncompressedByteSize
+          totalURLCount += 1
+          index += 1
+        }
+      }
+    }
+
+    val partitionCount = resultSetSerializables.length
+    val totalCompressedSizeMB = totalCompressedSize.toDouble/1024.0/1024.0
+    val totalUnCompressedSizeMB = totalUnCompressedSize.toDouble/1024.0/1024.0
+    log.info(
+      s"""printStatForSnowflakeResultSetRDD Total statistics:
+         | partitionCount=$partitionCount rowCount=$totalRowCount
+         | URLCount=$totalURLCount compressSizeMB=$totalCompressedSizeMB
+         | unCompressSizeMB=$totalUnCompressedSizeMB
+         |""".stripMargin.filter(_ >= ' '))
+
+    val aveCount = totalRowCount/partitionCount
+    val aveUrlCount = totalURLCount/partitionCount
+    val aveCompressSizeMB = totalCompressedSizeMB/partitionCount
+    val aveUnCompressSizeMB = totalUnCompressedSizeMB/partitionCount
+    log.info(
+      s"""printStatForSnowflakeResultSetRDD Average statistics per partition:
+         | rowCount=$aveCount URLCount=$aveUrlCount
+         | compressSizeMB=$aveCompressSizeMB unCompressSizeMB=$aveUnCompressSizeMB
+         |""".stripMargin.filter(_ >= ' '))
+  }
+
+  // Get data size and row count for first data chunk
+  private def getFirstChunkSize(
+      resultSetSerializable: SnowflakeResultSetSerializableV1): (Long, Long) =
+  {
+    val firstChunkData = resultSetSerializable.getFirstChunkStringData
+
+    if (firstChunkData == null || firstChunkData.length < 1) {
+      // There is no first chunk
+      (0, 0)
+    } else if (resultSetSerializable.getQueryResultFormat.equals(QueryResultFormat.ARROW)) {
+      // The first chunk is ARROW format
+      var rowCount: Long = 0
+      try {
+        val bytes = Base64.getDecoder.decode(firstChunkData)
+        val is = new ByteArrayInputStream(bytes)
+
+        val rootAllocator = new RootAllocator(Long.MaxValue)
+
+        val reader = new ArrowStreamReader(is, rootAllocator)
+        val root = reader.getVectorSchemaRoot
+        root.getFieldVectors
+        while (reader.loadNextBatch()) {
+          rowCount += root.getRowCount
+          root.clear
+        }
+
+        root.clear
+        reader.close
+        is.close
```

Comment:
These should be in a finally clause?

Reply:
You are right! I will move them to finally clause.